docs/ref/models/check-constraints.txt

```diff
@@ -0,0 +1,43 @@
+===========================
+Check constraints reference
+===========================
+
+.. module:: django.db.models.constraints
+
+.. currentmodule:: django.db.models
+
+.. versionadded:: 2.0
+
+The CheckConstraint class allows creating database check constraints. They can be added using the
+:attr:`Meta.constraints <django.db.models.Options.constraints>` option. This document
+explains the API references of :class:`CheckConstraint`.
+
+.. admonition:: Referencing built-in constraints
+
+    Constraints are defined in ``django.db.models.constraints``, but for convenience
+    they're imported into :mod:`django.db.models`. The standard convention is
+    to use ``from django.db import models`` and refer to the constraints as
+    ``models.CheckConstraint``.
+
+``CheckConstraint`` options
+===========================
+
+.. class:: CheckConstraint(constraint, name)
```

Comment:
I think it would be more natural if `name` was first, especially because `Q` objects can be arbitrarily complex.

Reply:
Do constraints really need an explicit, user-defined name?

Django autogenerates a name for indexes. I would expect it do the same for constraints. This suggests `CheckConstraint(constraint, name=None)`.

Reply:
(This could be retrofitted later.)

Reply:
That's a good point. The same algorithm can probably be used.

Reply:
Even if indexes use autogenerated names when no explicit ones are given, the mere presence of maintaining an autogenerated name even though the source for the generator (field names) can change is a pain. We can add autogeneration at some point, but taking it away is not really possible without too much trouble for our users.

The name attribute is last for `Index` as well. I think I'd leave it there.